plugins/modules/openssl_csr.py

```diff
@@ -478,796 +236,91 @@
-                    digest = cryptography.x509.SubjectKeyIdentifier.from_public_key(self.privatekey.public_key()).digest
-                    return ext.value.digest == digest
-                else:
-                    return ext.value.digest == self.subject_key_identifier
-            else:
-                return ext is None
-
-        def _check_authority_key_identifier(extensions):
-            ext = _find_extension(extensions, cryptography.x509.AuthorityKeyIdentifier)
-            if self.authority_key_identifier is not None or self.authority_cert_issuer is not None or self.authority_cert_serial_number is not None:
-                if not ext or ext.critical:
-                    return False
-                aci = None
-                csr_aci = None
-                if self.authority_cert_issuer is not None:
-                    aci = [str(cryptography_get_name(n)) for n in self.authority_cert_issuer]
-                if ext.value.authority_cert_issuer is not None:
-                    csr_aci = [str(n) for n in ext.value.authority_cert_issuer]
-                return (ext.value.key_identifier == self.authority_key_identifier
-                        and csr_aci == aci
-                        and ext.value.authority_cert_serial_number == self.authority_cert_serial_number)
-            else:
-                return ext is None
-
-        def _check_extensions(csr):
-            extensions = csr.extensions
-            return (_check_subjectAltName(extensions) and _check_keyUsage(extensions) and
-                    _check_extenededKeyUsage(extensions) and _check_basicConstraints(extensions) and
-                    _check_ocspMustStaple(extensions) and _check_subject_key_identifier(extensions) and
-                    _check_authority_key_identifier(extensions) and _check_nameConstraints(extensions))
-
-        def _check_signature(csr):
-            if not csr.is_signature_valid:
-                return False
-            # To check whether public key of CSR belongs to private key,
-            # encode both public keys and compare PEMs.
-            key_a = csr.public_key().public_bytes(
-                cryptography.hazmat.primitives.serialization.Encoding.PEM,
-                cryptography.hazmat.primitives.serialization.PublicFormat.SubjectPublicKeyInfo
-            )
-            key_b = self.privatekey.public_key().public_bytes(
-                cryptography.hazmat.primitives.serialization.Encoding.PEM,
-                cryptography.hazmat.primitives.serialization.PublicFormat.SubjectPublicKeyInfo
-            )
-            return key_a == key_b
-
-        try:
-            csr = load_certificate_request(self.path, backend='cryptography')
-        except Exception as dummy:
-            return False
-
-        return _check_subject(csr) and _check_extensions(csr) and _check_signature(csr)
-
-
 def main():
+    argument_spec, required_together, required_if, mutually_exclusive, required_one_of = get_csr_argument_spec()
```

Comment:
I added in 86758a2a097cb6cec7bb1ecf8307e8c3f3768ae0 a helper to prevent this. Once this PR is merged, I'll adjust it to handle the situation in #119 as well in #119 (so that PR should be merged second).